books.php

```diff
@@ -15,6 +15,85 @@ public static function getInstance()
         }
         return self::$instance;
     }
+    public function auth($login, $password)
+    {
+        $stmt = $this->link->prepare("SELECT id FROM users WHERE BINARY login = :login AND BINARY password = :password");
+        $stmt->execute([':login' => $login, ':password' => $password]);
+        if ($stmt->rowCount() > 0) {
+            return true;
+        }
+        return false;
+    }
+    public function add($data)
+    {
+        $i = 0;
+        $arrayTable = ['name' => 0, 'genre' => 0, 'pages' => 0];
+        if ((count($data, COUNT_RECURSIVE) - count($data)) > 0) {
+            foreach ($data as $key => $array1) {
+                var_dump(implode("','", array_intersect_key($array1, $arrayTable)));
+                $stmt = $this->link->prepare("INSERT INTO `book`(`name`, `genre`, `pages`) VALUES (" . "'" . implode("','", array_intersect_key($array1, $arrayTable)) . "'" . ")");
+                $stmt->execute();
+                $i = $stmt->rowCount() + $i;
+            }
+            return $i;
+        }
+        $stmt = $this->link->prepare("INSERT INTO `book`(`name`, `genre`, `pages`) VALUES (" . "'" . implode("','", array_intersect_key($data, $arrayTable)) . "'" . ")");
+        $stmt->execute();
+        var_dump($stmt->rowCount());
+    }
+    public function update($data, $url)
+    {
+        if (preg_match("[^/api/books(|/)$]", $url)) {

```

Comment:
You don't need to put here such logic.
This method should just update books table.
All such logic should be upper in top level.
Function should obtain `$id=null` as parameter.

Reply:
In case if request `update books` without id. It should keep array of books with id in each object.
In top level you need to obtain all books id and provide this IDs in foreach to `update()` method.
Or can just ask clients to send only requests in predefined format like:
```
{id1:{"pages":345},id2:{"name":"test2"}}
```